xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -667,6 +668,12 @@ def parse_optional_group(self) -> FormatDirective:
                 if anchor is not None:
                     self.raise_error("An optional group can only have one anchor.")
                 anchor = then_elements[-1]
+
+        if self.parse_optional_punctuation(":"):
+            self.parse_punctuation("(")
+            while not self.parse_optional_punctuation(")"):
+                else_elements += (self.parse_format_directive(),)
```

Comment:
From my understanding of the docs, the anchor could also be in `else_elements`

Reply:
ah, sorry I didn't see your previous comment, seems on par for MLIR documentation. I'd be happy to either go with the version here or with a version which allows an anchor in the `else_elements`.